internal/mongodb/mongodb.go

```diff
@@ -0,0 +1,334 @@
+// SPDX-FileCopyrightText: 2023 Risk.Ident GmbH <contact@riskident.com>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+package mongodb
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+)
+
+var (
+	ErrNotFound = errors.New("not found")
+	ErrNotOK    = errors.New("not ok")
+
+	AppName = "terraform-provider-mongodb-driver"
+)
+
+type Client struct {
+	uri         string
+	credentials Credentials
+	connectOnce sync.Once
+}
+
+type Credentials struct {
+	Username string
+	Password string
+}
+
+func New(uri string, cred Credentials) *Client {
+	return &Client{
+		uri:         uri,
+		credentials: cred,
+	}
+}
+
+func (c *Client) connect(ctx context.Context) (client *mongo.Client, connectErr error) {
+	c.connectOnce.Do(func() {
+		ctx, cancel := context.WithTimeout(ctx, 20*time.Second)
```

Comment:
nit: Maybe this timeout should be configurable?

Reply:
Removed timeouts from `mongodb` package and instead provided them from the `provider` package. Let's users define timeouts "the de-facto terraform way"

https://developer.hashicorp.com/terraform/plugin/framework/resources/timeouts